File: ios/Plugin/DialerPlugin.swift
import Foundation
import Capacitor

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(DialerPlugin)
public class DialerPlugin: CAPPlugin {
    private let implementation = Dialer()

    // @objc func echo(_ call: CAPPluginCall) {
    //     let value = call.getString("value") ?? ""
    //     call.resolve([
    //         "value": implementation.echo(value)
    //     ])
    // }
     @objc func openDialer(_ call: CAPPluginCall) {
        guard let phoneNumber = call.getString("phoneNumber") else {
            call.reject("Must provide a phone number")
            return
        }

        guard let url = URL(string: "tel://\(phoneNumber)"), UIApplication.shared.canOpenURL(url) else {
            call.reject("Unable to open phone dialer")
            return
        }

        UIApplication.shared.open(url, options: [:], completionHandler: nil)
        call.resolve()
    }
}

File: android/src/main/java/com/xconcepts/plugin/dialer/DialerPlugin.java
package com.xconcepts.plugin.dialer;

import android.content.Intent;
import android.net.Uri;

import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

@CapacitorPlugin(name = "Dialer")
public class DialerPlugin extends Plugin {

    private Dialer implementation = new Dialer();

    @PluginMethod
    public void openDialer(PluginCall call) {
        // String value = call.getString("value");

        // JSObject ret = new JSObject();
        // ret.put("value", implementation.echo(value));
        // call.resolve(ret);
        String phoneNumber = call.getString("phoneNumber");
        if (phoneNumber == null) {
            call.reject("Must provide a phone number");
            return;
        }

        Intent intent = new Intent(Intent.ACTION_DIAL);
        intent.setData(Uri.parse("tel:" + phoneNumber));
        getActivity().startActivity(intent);

        call.resolve(new JSObject());
    }
}
